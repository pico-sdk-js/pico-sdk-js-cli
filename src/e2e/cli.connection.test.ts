import { describe, it } from '@jest/globals';
import psjRunner from './psjRunner';

describe('PSJ Connection Scenarios', () => {
    describe('command line args', () => {
        describe('--help', () => {
            it('shows help', async () => {
                // prettier-ignore
                await psjRunner()
                    .start(['--help'])
                    .assertExitCode(0)
                    .assertSnapshot();
            });
        });

        describe('--connection', () => {
            it('connects to /dev/ttyACM0', async () => {
                // prettier-ignore
                await psjRunner()
                    .start(['--skip-header', '--connection', '/dev/ttyACM0'])
                    .assertSnapshot();
            });

            it('searches for connection', async () => {
                // prettier-ignore
                await psjRunner()
                    .start(['--skip-header', '--connection', 'auto'])
                    .assertSnapshot();
            });

            it('fails connection to invalid device', async () => {
                // prettier-ignore
                await psjRunner()
                    .start(['--skip-header', '--connection', '/dev/ttyACM999'])
                    .assertSnapshot();
            });
        });
    });

    describe('.exit', () => {
        it('in REPL quits process', async () => {
            // prettier-ignore
            await psjRunner()
                .start(['--skip-header'])
                .command('.exit')
                .assertSnapshot()
                .assertExitCode(0);
        });
    });
});
