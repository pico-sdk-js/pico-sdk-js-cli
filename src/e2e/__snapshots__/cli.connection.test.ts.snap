// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PSJ Connection Scenarios .exit in REPL quits process: stdOut 1`] = `
"Searching for device running Pico-SDK-JS engine...
Connected to Pico-SDK-JS Engine v0.0.3 at '/dev/ttyACM0'
> .exit
ERR: Connection to Pico-SDK-JS engine lost."
`;

exports[`PSJ Connection Scenarios .ls shows error when not connected: stdOut 1`] = `
"Searching for device running Pico-SDK-JS engine...
Connected to Pico-SDK-JS Engine v0.0.3 at '/dev/ttyACM0'
> .ls
total 0 file(s)
> "
`;

exports[`PSJ Connection Scenarios command line args --connection connects to /dev/ttyACM0: stdOut 1`] = `
"Connecting to serial device at /dev/ttyACM0
Connected to Pico-SDK-JS Engine v0.0.3 at '/dev/ttyACM0'
> "
`;

exports[`PSJ Connection Scenarios command line args --connection fails connection to invalid device: stdOut 1`] = `
"Connecting to serial device at /dev/ttyACM999
ERR: Error: Error: No such file or directory, cannot open /dev/ttyACM999"
`;

exports[`PSJ Connection Scenarios command line args --connection searches for connection: stdOut 1`] = `
"Searching for device running Pico-SDK-JS engine...
Connected to Pico-SDK-JS Engine v0.0.3 at '/dev/ttyACM0'
> "
`;

exports[`PSJ Connection Scenarios command line args --help shows help: stdOut 1`] = `
"psj
Starts the REPL server to connect to the Raspberry Pi Pico
Options:
      --help             Show help                                     [boolean]
      --version          Show version number                           [boolean]
      --skip-header      Do not output the process header.
                                                      [boolean] [default: false]
      --log-level, --ll  Sets the log level of the output.
     [string] [choices: "error", "warning", "debug", "trace"] [default: "error"]
  -c, --connection       Specifies the target device to connect to: "local" for
                         a local process, "auto" for the first available device,
                         or a specific serial port path (e.g., /dev/tty.XXX or
                         COM1).                       [string] [default: "auto"]
For more information, check out our docs on https://pico-sdk-js.github.io/"
`;
